doc/md/versioned/troubleshoot.mdx

```diff
@@ -0,0 +1,578 @@
+This can be done manually or with the help of `atlas schema apply` / `atlas schema diff` commands. Both commands know
+how to read a database state from a migration directory and compare it with a state read from a database. The only
+difference is, that `atlas schema apply` can then automatically apply the changes to the database, while
+`atlas schema diff` will print the SQL on screen for you to check run by hand. You can utilize the `atlas schema status`
+command to get information about the currently applied migration version.
+
+##### atlas schema diff
+
+<Tabs
+defaultValue="dir"
+values={[
+{label: 'Migration Directory', value: 'dir'},
+{label: 'HCL File', value: 'hcl'},
+{label: 'Another Database', value: 'db'},
+]}>
+<TabItem value="dir">
+
+Compare the state of the `from` database (the one with the schema drift), to the desired state given in the migration
+directory. Note the query parameter `version` in the connection string to the migration directory, specifying the
+version to consider the current state of the schema.
+
+```shell
+atlas schema diff \
+  --from "driver://user:pass@host:port/db" \
+  --to "file:///path/to/migrations/dir?format=atlas&version=myTargetVersion" \
+  --dev-url "driver://user:pass@host:port/dev_db"
+```
+
+</TabItem>
+<TabItem value="hcl">
+
+Compare the state of the `from` database (the one with the schema drift), to the desired state given in one or more HCL
+files.
+
+```shell
+atlas schema diff \
+  --from "driver://user:pass@host:port/db" \
+  --to "file:///path/to/schema_1.hcl" \
+  --to "file:///path/to/schema_2.hcl" \
+  --dev-url "driver://user:pass@host:port/dev_db"
+```
+
+</TabItem>
+<TabItem value="db">
+
+Compare the state of the `from` database (the one with the schema drift), to the desired state given by another database
+(e.g. a local dev copy of the schema).
+
+```shell
+atlas schema diff \
+  --from "driver://user:pass@host:port/current" \
+  --to "driver://user:pass@host:port/desired"
+```
+
+</TabItem>
+</Tabs>
+
+Once you apply the SQL shown by `atlas migrate diff` onto your database, the schema drift should have vanished, and you
+can retry migration applying.
+
+#### atlas schema apply
+
+Instead of only showing the SQL on the screen, you can immediately let Atlas apply it (Atlas will prompt for
+confirmation first).
+
+<Tabs
+defaultValue="dir"
+values={[
+{label: 'Migration Directory', value: 'dir'},
+{label: 'HCL File', value: 'hcl'},
+{label: 'Another Database', value: 'db'},
+]}>
+<TabItem value="dir">
+
+Migrate the state of the `url` database (the one with the schema drift), to the desired state given in the migration
+directory. Note the query parameter `version` in the connection string to the migration directory, specifying the
+version to consider the current state of the schema.
+
+```shell
+atlas schema apply \
+  --url "driver://user:pass@host:port/db" \
+  --to "file:///path/to/migrations/dir?format=atlas&version=myTargetVersion" \
```

Comment:
```suggestion
  --to "file:///path/to/migrations/dir?version=myTargetVersion" \
```
Mention the query-string parameters above instead, if you like.

Reply:
Does not make sense since troubleshooting as for now is for out own execution.